### DISCUSSION TITLE
Why connect number can big than poolSize?

### DISCUSSION THREAD

**Discussion by codergma** [redis/go-redis#1808] [Q&A] [answered]
When the pool is full, new connections are still created in newConn function. why?
![clipboard_image_1625057768974](https://user-images.githubusercontent.com/11986423/123964219-ccb3b680-d9e5-11eb-857f-b4a205fe5a83.png)

**Comment by monkey92t** [comment, marked as answer]
You can see here

https://github.com/go-redis/redis/blob/dd4b7eb3b41c39c4f74ca016f09c55770df449c4/internal/pool/pool.go#L270-L304

Under normal circumstances, it needs to enter a queue first (length is poolsize). In special cases (such as PubSub), it may create a connection that is not in the connection pool (pooled=false)

**Reply by codergma** [reply]
thanks.
